jujugui/static/gui/src/app/components/generic-input/test-generic-input.js

```diff
@@ -352,6 +360,30 @@ describe('GenericInput', function() {
     assert.deepEqual(output.props.className, 'generic-input has-error');
   });
 
+  it('adds an error icon with inlineErrorIcon is set', () => {
+    const renderer = jsTestUtils.shallowRender(
+      <juju.components.GenericInput
+        disabled={false}
+        inlineErrorIcon={true}
+        placeholder="placeholder"
+        required={true}
+        ref="test"
+        validate={[{
+          regex: /\S+/,
+          error: 'This field is required.'
+        }]} />, true);
+    const instance = renderer.getMountedInstance();
+    instance.refs = {field: {value: ''}};
+    let output = renderer.getRenderOutput();
+    output.props.children[1].props.onBlur();
+    output = renderer.getRenderOutput();
+    const expected = (<juju.components.SvgIcon
```

Comment:
Someone once suggested me that just `expected` is usually not enough, and we should add some context, like `expectedOutput`. Take it or leave it.